osidb/mixins.py

```diff
@@ -462,3 +462,37 @@ def _validate_acl_identical_to_parent_flaw(self):
 
     class Meta:
         abstract = True
+
+
+class TextChoicesComparisonMixin:
+    """
+    mixin to extend the models.TextChoices classes
+    making them comparable with the standard operators
+    """
+
+    @classmethod
+    def get_choices(cls):
+        """
+        get processed choices
+        """
+        return [choice[0] for choice in cls.choices]
+
+    @property
+    def weight(self):
+        """
+        weight of the instance for the comparison
+        defined by the order of the definition of the choices
+        """
+        return self.get_choices().index(str(self))
+
+    def __lt__(self, other):
+        return self.weight < other.weight
+
+    def __gt__(self, other):
+        return self.weight > other.weight
+
+    def __le__(self, other):
+        return self == other or self.__lt__(other)
+
+    def __ge__(self, other):
+        return self == other or self.__gt__(other)
```

Comment:
Please see and use `IntegerChoices` https://docs.djangoproject.com/en/4.2/ref/models/fields/#enumeration-types

Reply:
This reasonable. I will have a look Thank you

Reply:
@Elkasitu this does not seem to work the way we would like to. It seems to only be a named integer enumeration while we really need strings. The string conversions then result in strings like `1` instead of the actual name. Even the last example in the documentation with the `Answer` class works the same way.

Reply:
But anyway thank you for pointing me to this. I completely forgot that it exists :angel: